### ISSUE TITLE
vmx_data configuration for vsphere-iso builder

### ISSUE THREAD

**Issue by minhluantran017** [hashicorp/packer#9188]
Hi,

Previously I used `vmware-iso` builder on ESXi host (with ESXi modification). I have the section for `vmx_data` in template to set my own value.

But when I change to `vsphere-iso` (no longer need modification, it's great), it turns out not having the `vmx_data` configuration in template.

I open this issue to request a `vmx_data` configuration in `vsphere-iso` builder.

Thanks.

**Comment by SwampDragons** [comment]
The field has a different name in the vsphere-iso builder: `configuration_parameters` https://www.packer.io/docs/builders/vsphere-iso.html#configuration_parameters. Sorry for the confusion -- we only merged the vsphere iso builder with the packer core a few months ago and we're still cleaning up inconsistencies in option names between the vsphere and vmware builders.

**Comment by minhluantran017** [comment]
It's great to know that. Thank @SwampDragons 😃

**Comment by minhluantran017** [comment]
Hi @SwampDragons ,
I don't know if anything is configured incorrectly but the `configuration_parameters` doesn't work for me. Eg, I put this to enable **nested virtualization** enabled. It works on `vmware-iso` but not on `vsphere-iso`:
```
"configuration_parameters": [
                "ethernet0.networkName": "{{user `vsphere_network`}}",
                "ethernet0.present": "TRUE",
                "ethernet0.startConnected": "TRUE",
                "featMask.vm.hv.capable": "Min:1",
                "vhv.enable": "true"
],
```

**Comment by SwampDragons** [comment]
you probably just mistyped for the example here, but it should use {} not []: 

"configuration_parameters": {
                "ethernet0.networkName": "{{user `vsphere_network`}}",
                "ethernet0.present": "TRUE",
                "ethernet0.startConnected": "TRUE",
                "featMask.vm.hv.capable": "Min:1",
                "vhv.enable": "true"
},

If the issue is nested virtualixation, does https://www.packer.io/docs/builders/vsphere-iso.html#nestedhv not work?

**Comment by username-is-already-taken2** [comment]
Hi @SwampDragons, Thanks for supporting this issue.  I've been able to use `configuration_parameters` would you happen to know if `vmx_data_post` is mapped to different parameter?  no worries if not.  I find it hard to follow the go code to look myself

**Comment by SwampDragons** [comment]
I don't think there's an analogue for vmx_data_post in the VSphere builder, currently, but it probably wouldn't be too hard to add. I'd prioritize reviewing a PR adding that option, but if you don't feel like you have the golang chops to do it yourself, do you think you could open a new feature request issue so it doesn't get lost in this ticket?

**Comment by SwampDragons** [comment]
@minhluantran017 any updates on whether the above advice helped?

**Comment by tranluanms** [comment]
@SwampDragons Thanks for the suggestion. I did find the option NestedHV works fine, not too much work on other vmx_data configurations. So the issue is resolved.

**Comment by minhluantran017** [comment]
@SwampDragons thanks for the suggestion. I found NestedHV works fine. So I did not have too much work on other vmx_data configs. The issue can be closed now. :)
